driver-core/src/main/java/com/datastax/driver/core/AggregateMetadata.java

```diff
@@ -101,11 +106,29 @@ static AggregateMetadata build(KeyspaceMetadata ksm, Row row, VersionNumber vers
         if (version.getMajor() >= 3) {
             stateType = DataTypeCqlNameParser.parse(stateTypeName, cluster, ksm.getName(), ksm.userTypes, null, false, false);
             String rawInitCond = row.getString("initcond");
-            initCond = rawInitCond == null ? null : codecRegistry.codecFor(stateType).parse(rawInitCond);
+            if (rawInitCond == null) {
+                initCond = null;
+            } else {
+                try {
+                    initCond = codecRegistry.codecFor(stateType).parse(rawInitCond);
```

Comment:
To sum up:

> CREATE AGGREGATE ks_1.ag() SFUNC id STYPE 'org.apache.cassandra.db.marshal.DynamicCompositeType(s=>org.apache.cassandra.db.marshal.UTF8Type,i=>org.apache.cassandra.db.marshal.Int32Type)' INITCOND 0x80730003666f6f00806900040000002000;

On 2.2.X, 3.0.3, 3.3 yields:

> Invalid component representation: 0x80730003666f6f00806900040000002000)

On 3.0.4, 3.4 it works.


> CREATE AGGREGATE ks_1.ag() SFUNC id STYPE 'org.apache.cassandra.db.marshal.DynamicCompositeType(s=>org.apache.cassandra.db.marshal.UTF8Type,i=>org.apache.cassandra.db.marshal.Int32Type)' INITCOND 's@foo:i@32';

On 3.0.3, 3.3 yields:

> java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.apache.cassandra.exceptions.SyntaxException: Failed parsing CQL term: [s@foo:i@32] reason: SyntaxException line 1:1 no viable alternative at character '@'

On 2.2.5, 3.0.4, and 3.4 it works.